public/app/plugins/panel/flamegraph/components/FlameGraphContainer.tsx

```diff
@@ -97,27 +102,31 @@ const FlameGraphContainer = (props: Props) => {
                       grafana_version: config.buildInfo.version,
                     });
                     setSearch(symbol);
-                    // Reset selected level in flamegraph when selecting row in top table
-                    setRangeMin(0);
-                    setRangeMax(1);
+                    resetFocus();
                   }
                 }}
+                height={selectedView === SelectedView.TopTable ? 600 : undefined}
               />
             )}
 
             {selectedView !== SelectedView.TopTable && (
               <FlameGraph
                 data={dataContainer}
-                levels={levels}
                 rangeMin={rangeMin}
                 rangeMax={rangeMax}
                 search={search}
                 setRangeMin={setRangeMin}
                 setRangeMax={setRangeMax}
-                selectedView={selectedView}
-                onItemFocused={(itemIndex) => setFocusedItemIndex(itemIndex)}
-                focusedItemIndex={focusedItemIndex}
+                onItemFocused={(data) => setFocusedItemData(data)}
+                focusedItemData={focusedItemData}
                 textAlign={textAlign}
+                sandwichItem={sandwichItem}
+                onSandwich={(label: string) => {
+                  resetFocus();
```

Comment:
So I think going to merge view with the same focus does not make much sense. Like we rearrange the graph a lot at that point and the items shift around also because of merging subtrees the sizes of items do not match so keeping the focus may lead to nonsensical view/slice of the graph imho.

Reply:
Yeah, I tend to agree with this thanks for walking through it